docs/content/guides/dagster/troubleshooting.mdx

```diff
@@ -0,0 +1,120 @@
+- Do I have any potential for flakey failures? How do I build retries into my pipeline?
+  [How to recover from flaky failures using retries](concepts/ops-jobs-graphs/op-retries#op-retries)
+- How to diagnose failures? What types of logs would it be helpful to include in my pipeline on portions that might be more sensitive?[Learn about Dagster's built-in loggers](/concepts/logging/loggers#using-built-in-loggers) 
+- What alerts in slack or email are important enough to trigger? When is it time to jump in and see what the problem is?[Setting up run failure alerts in Dagster with Sensors](/concepts/partitions-schedules-sensors/sensors#run-failure-sensor)
+
+
+## Op or Asset Failures
+
+One of the most common issues that users will hit in Dagster is either an Asset or Op failure. 
+
+#### What went wrong?
+
+Dagit can be used to review the [loggers](/concepts/logging/loggers#using-loggers) to review the stacktrace to investigate what has failed. A few common cases:
+- The Asset or Op could have received some bad data. For more information see [Validating Data with Dagster](/guides/dagster/dagster_type_factories)
+- An external service might have failed, retries in Dagster can be helpful to mitigate this. For more information see [How to recover from flaky failures using retries](concepts/ops-jobs-graphs/op-retries#op-retries)
+- If a non-python sub-process was used, the raw compute logs can be helpful (for more information click here)
+- If something is going wrong with another python library, linking the logger to capture some of the events 
+
+Helpful hint:
+[Exception chaining may be helpful](https://docs.python.org/3.9/tutorial/errors.html#exception-chaining)
+
+---
+
+### Observability 
+
+Dagster provides built-in logger as well an extendible logging system that can be customized. 
+
+Dagster has structured event, compute, agent health and pod logs that can be published. These logs can help dig into what is not working.
+
+
+<!-- ![Viewing the cereals asset in Dagit](/images/guides/troubleshooting/agents.png) -->
+
+<Image
+alt="Viewing the agent console in Dagit"
+src="/images/guides/troubleshooting/agents.png"
+width={3574}
+height={1962}
+/>
+
+Some helpful logs that are available when using Dagster locally or a Hybrid deployment:
+
+- Agent status is available in the Dagit UI which displays if there is a heartbeat and direct access to raw logs is available 
+- Code server status and logs are directly available 
+- Step & Run worker status is available via agent checks and through the event logs as well as raw logs via the compute logs 
+
+#### Deployment Troubleshooting
+
+Agent and pod logs can be a useful tool for resolving Dagster deployment issues. 
+
+[For more information on Dagster Cloud Hybrid Deploymements](/dagster-cloud/deployment/agents)
+
+---
+
+
+## Sensing Failures 
+
+#### Setting up sensors 
```

Comment:
🔧 Both subheadings in this section should be H3, not H4, as the heading that contains them is an H2.